WordPress/Classes/Utility/ContextManager.m

```diff
@@ -188,6 +188,23 @@ - (BOOL)obtainPermanentIDForObject:(NSManagedObject *)managedObject
     return YES;
 }
 
+- (void)mergeChanges:(NSManagedObjectContext *)context fromContextDidSaveNotification:(NSNotification *)notification
+{
+    [context performBlock:^{
+        // Fault-in updated objects before a merge to avoid any internal inconsistency errors later.
+        // Based on old solution referenced here: http://www.mlsite.net/blog/?p=518
+        NSSet* updates = [notification.userInfo objectForKey:NSUpdatedObjectsKey];
+        for (NSManagedObject *object in updates) {
+            if ([object isFault]) {
+                // Force a fault-in of the object's key-values
+                [[context objectWithID:[object objectID]] willAccessValueForKey:nil];
```

Comment:
Since the **object** itself may belong to another NSManagedObjectContext, it might be a good idea to reload the entity in the **context** that's being merged. I'm not sure what would be the effect of `willAccessValueForKey`, when crossing threads.

Something like this perhaps?

```
       NSArray* updates = [notification.userInfo[NSUpdatedObjectsKey] allObjects];
        for (NSManagedObject* object in updates) {
            NSManagedObject* localMO = [context existingObjectWithID: object.objectID error:nil];
            if (localMO.isFault) {
                [localMO willAccessValueForKey:nil];
            }
        }
```